src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -257,6 +271,7 @@ function WebGLShadowMap( _renderer, _objects, _capabilities ) {
 
 				materialsForVariant = {};
 				_materialCache[ keyA ] = materialsForVariant;
+				result.addEventListener( 'dispose', onDisposeCustomMaterial );
```

Comment:
This line adds an event listener to depth materials. Custom depth materials can be cleared in app level. However, how is `dispose()` going to be called for internal depth materials?

Reply:
It's not but at most this will mean one event listener gets added to the internal depth materials which seemed fine to me. We could add logic for preventing events from being added to the internal materials but it didn't seem worth it and simpler not to.

Regarding how the material subvarients for the internal materials are cleaned out that's what the second dispose listener I mentioned will be needed for which I was planning to add in another PR.

Reply:
Um, I guess it would be helpful to see your other PR so it's possible to understand the complete solution.

Reply:
Got it -- I thought it might be easier to see each solution if it was broken up. Just added it in this PR so now the case of the mesh materials being disposed is handled as well as the custom shadow materials.

Reply:
@Mugen87 Any thoughts here? It would be nice to address known memory leaks so they don't become forgotten.